payments-core/src/test/java/com/stripe/android/payments/core/injection/WeakMapInjectorRegistryTest.kt

```diff
@@ -0,0 +1,65 @@
+package com.stripe.android.payments.core.injection
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.mock
+import org.robolectric.RobolectricTestRunner
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+@RunWith(RobolectricTestRunner::class)
+@ExperimentalCoroutinesApi
+class WeakMapInjectorRegistryTest {
+
+    @Before
+    fun clearStaticCache() {
+        WeakMapInjectorRegistry.staticCache.clear()
+    }
+
+    @Test
+    fun verifyRegistryRetrievesCorrectObject() {
+        val injector1 = mock<Injector>()
+        val keyForInjector1 = WeakMapInjectorRegistry.nextKey()
+
+        val injector2 = mock<Injector>()
+        val keyForInjector2 = WeakMapInjectorRegistry.nextKey()
+
+        WeakMapInjectorRegistry.register(injector1, keyForInjector1)
+        WeakMapInjectorRegistry.register(injector2, keyForInjector2)
+
+        assertEquals(injector1, WeakMapInjectorRegistry.retrieve(keyForInjector1))
+        assertEquals(injector2, WeakMapInjectorRegistry.retrieve(keyForInjector2))
+        assertNotEquals(
+            WeakMapInjectorRegistry.retrieve(keyForInjector1),
+            WeakMapInjectorRegistry.retrieve(keyForInjector2)
+        )
+    }
+
+    @Suppress("UNUSED_VALUE")
+    @Test
+    fun verifyCacheIsClearedOnceWeakReferenceIsDeReferenced() {
+        var injector1: Injector? = mock()
+        val keyForInjector1 = WeakMapInjectorRegistry.nextKey()
+        WeakMapInjectorRegistry.register(injector1!!, keyForInjector1)
+
+        assertNotNull(WeakMapInjectorRegistry.retrieve(keyForInjector1))
+
+        var retry = 10
```

Comment:
the weak reference is gone after one `System.gc()` call in my local env. I'm not sure if 10 retries is good enough to trigger it in any JVM and open to suggestions on improving this test(or not testing it at all)

Reply:
I don't have an idea for a better test, so we can leave it and if it ends up being too flaky we can try to find a better way.